sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -335,14 +410,21 @@ private[sbtplugin] object ScalaJSPluginInternal {
         ((crossTarget in fullOptJS).value /
             ((moduleName in fullOptJS).value + "-opt.js")),
 
-      scalaJSLinkerConfig in fullOptJS ~= { prevConfig =>
+      scalaJSLinkerConfig in linkJSProd ~= { prevConfig =>
```

Comment:
This is the most annoying one, as it is a very real use case for people to tweak `scalaJSLinkerConfig in (Compile, fullOptJS)`. We absolutely must not throw away their configuration, *especially* since it also affects the result of the legacy keys themselves.

So we should still have here `in fullOptJS`, and after that add
```scala
scalaJSLinkerConfig in linkJSDev := (scalaJSLinkerConfig in fastOptJS).value,
scalaJSLinkerConfig in linkJSProd := (scalaJSLinkerConfig in fullOptJS).value,
```